addons/storysource/src/loader/generate-helpers.js

```diff
@@ -28,20 +28,25 @@ function generateSourceWithoutUglyComments(source, { comments, uglyCommentsRegex
 }
 
 function prettifyCode(source, { prettierConfig, parser, filepath }) {
-  let config = prettierConfig;
-
-  if (!config.parser) {
-    if (parser) {
-      config = {
-        ...prettierConfig,
-        parser: parser === 'javascript' ? 'babylon' : parser,
-      };
-    } else if (filepath) {
-      config = {
-        ...prettierConfig,
-        filepath,
-      };
-    }
+  if (prettierConfig.parser) return prettier.format(source, prettierConfig);
+
+  let config = {};
+
+  if (parser) {
+    config = {
+      ...prettierConfig,
+      parser: parser === 'javascript' ? 'babylon' : parser,
```

Comment:
let's put `...prettierConfig` after this line, so that it wouldn't override explicitly passed `prettierConfig.parser`

Reply:
[I made a slight change](https://github.com/storybooks/storybook/pull/3660/files#diff-4a68ffd7414441778cfa050339c5a114R31) to the beginning of this function.
When `prettierConfig.parser` is explicitly set `return` will be done at this point

> if (prettierConfig.parser) return prettier.format(source, prettierConfig);

After this processing, shoud not set prettierConfig.parser.
So I do not think it is necessary to change it

Reply:
But, if this code is not good, I think it can try changes.

Reply:
I don't feel really good about this double `prettier.format` call in different logical branches. Let's maybe revert it to how it was, with `if (!prettier config.parser)` check?

Reply:
OK, I understood.
Many thanks for having you review it over and over. And, sorry...

I reverted before commit.
After, modified to make parser or filepath always specified
If the parser and file path are empty, set the parser babylon